src/openzeppelin/access/accesscontrol/library.cairo

```diff
@@ -10,169 +10,147 @@ from starkware.cairo.common.bool import TRUE, FALSE
+        role: felt
+    ) {
+        alloc_locals;
+        let (caller) = get_caller_address();
+        let (authorized) = has_role(role, caller);
+        with_attr error_message("AccessControl: caller is missing role {role}") {
+            assert authorized = TRUE;
+        }
+        return ();
+    }
+
+    //
+    // Getters
+    //
+
+    func has_role{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        role: felt, user: felt
+    ) -> (has_role: felt) {
+        let (authorized: felt) = AccessControl_role_member.read(role, user);
+        return (authorized,);
+    }
+
+    func get_role_admin{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        role: felt
+    ) -> (admin: felt) {
+        let (admin: felt) = AccessControl_role_admin.read(role);
+        return (admin=admin);
+    }
+
+    //
+    // Externals
+    //
+
+    func grant_role{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        role: felt, user: felt
+    ) {
+        let (admin: felt) = get_role_admin(role);
+        assert_only_role(admin);
+        _grant_role(role, user);
+        return ();
+    }
+
+    func revoke_role{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        role: felt, user: felt
+    ) {
+        let (admin: felt) = get_role_admin(role);
+        assert_only_role(admin);
+        _revoke_role(role, user);
+        return ();
+    }
+
+    func renounce_role{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        role: felt, user: felt
+    ) {
+        let (caller: felt) = get_caller_address();
+        with_attr error_message("AccessControl: can only renounce roles for self") {
+            assert user = caller;
+        }
+        _revoke_role(role, user);
+        return ();
+    }
+
+    //
+    // Unprotected
+    //
+
+    func _grant_role{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        role: felt, user: felt
+    ) {
+        let (user_has_role: felt) = has_role(role, user);
+        if (user_has_role == FALSE) {
+            let (caller: felt) = get_caller_address();
+            AccessControl_role_member.write(role, user, TRUE);
+            RoleGranted.emit(role, user, caller);
+            return ();
+        }
+        return ();
+    }
+
+    func _revoke_role{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        role: felt, user: felt
+    ) {
+        let (user_has_role: felt) = has_role(role, user);
+        if (user_has_role == TRUE) {
+            let (caller: felt) = get_caller_address();
+            AccessControl_role_member.write(role, user, FALSE);
+            RoleRevoked.emit(role, user, caller);
+            return ();
+        }
+        return ();
+    }
+
+    func _set_role_admin{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        role: felt, admin_role: felt
```

Comment:
mh this looks like a good opportunity to find and document differences between our coding style and the standard one, like not inlining implicit params

Reply:
A pinned thread in Slack for listing format guidelines would be helpful?